[check-webkit-style] Support branches

JonWBedard opened the pull request:
#### 64078f4d67054e7bdd5a617cc0bb596a3d327ac7
<pre>
[check-webkit-style] Support branches
<a href="https://bugs.webkit.org/show_bug.cgi?id=242664">https://bugs.webkit.org/show_bug.cgi?id=242664</a>
&lt;rdar://96914421&gt;

Reviewed by Stephanie Lewis.

* Tools/Scripts/libraries/webkitscmpy/setup.py: Bump version.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/__init__.py: Ditto.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/branch.py:
(Branch.editable): Check all source remotes, not just origin.
(Branch.branch_point): Provide a limit.
* Tools/Scripts/webkitpy/common/checkout/scm/git.py:
(Git.merge_base): Pass find_branch flag.
(Git.changed_files): Ditto.
(Git.create_patch): Ditto.
(Git.remote_merge_base): Ditto.
(Git.remote_branch_ref): If requested, attempt to find the branch point of the
current branch. This allows for development from branches other than main.
* Tools/Scripts/webkitpy/common/checkout/scm/scm.py:
(SCM.changed_files): Pass find_branch flag.
(SCM.create_patch): Ditto.
* Tools/Scripts/webkitpy/common/checkout/scm/scm_mock.py:
(MockSCM.changed_files): Pass find_branch flag.
(MockSCM.create_patch): Ditto.
* Tools/Scripts/webkitpy/common/checkout/scm/svn.py:
(SVN.changed_files): Pass find_branch flag.
(SVN.create_patch): Ditto.
* Tools/Scripts/webkitpy/style/main.py:
(CheckWebKitStyle.main): Check style from the branch point, not main.
* Tools/Scripts/webkitpy/tool/steps/abstractstep.py:
(AbstractStep): Create a patch from the branch point.
* Tools/Scripts/webkitpy/w3c/test_exporter_unittest.py:
(TestExporterTest.MockGit.create_patch): Pass find_branch flag.

Canonical link: <a href="https://commits.webkit.org/252647@main">https://commits.webkit.org/252647@main</a>
</pre>

sundiamonde commented on Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/branch.py:
```diff
@@ -62,18 +62,21 @@ def editable(cls, branch, repository=None):
         commit = repository.commit(include_log=False, include_identifier=False)
         while cls.editable(commit.branch, repository=repository):
             cnt += 1
+            if limit and cnt > limit:
```
why do we need a limit? expensive to run? can get too deeply nested?

JonWBedard replied:
We want a limit because it's possible for folks to have poorly configured checkouts and for this function to find hundreds or thousands of commits. If we've gone past about a dozen, something has probably gone wrong.

sundiamonde commented on Tools/Scripts/webkitpy/common/checkout/scm/git.py:
```diff
@@ -548,22 +549,22 @@ def delete_branch(self, branch_name):
+                for remote in repository.source_remotes():
+                    if branch_point.branch in repository.branches_for(remote=remote, cached=True):
+                        return 'refs/remotes/{}/{}'.format(remote, branch_point.branch)
+        for ref in ['refs/remotes/origin/mainy', 'refs/remotes/origin/master']:
```
typo mainy -> main

JonWBedard replied:
Oh oops, this was for testing anyways, thanks for catching that.

JonWBedard committed:
[check-webkit-style] Support branches
https://bugs.webkit.org/show_bug.cgi?id=242664
<rdar://96914421>

Reviewed by Stephanie Lewis.

* Tools/Scripts/libraries/webkitscmpy/setup.py: Bump version.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/__init__.py: Ditto.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/branch.py:
(Branch.editable): Check all source remotes, not just origin.
(Branch.branch_point): Provide a limit.
* Tools/Scripts/webkitpy/common/checkout/scm/git.py:
(Git.merge_base): Pass find_branch flag.
(Git.changed_files): Ditto.
(Git.create_patch): Ditto.
(Git.remote_merge_base): Ditto.
(Git.remote_branch_ref): If requested, attempt to find the branch point of the
current branch. This allows for development from branches other than main.
* Tools/Scripts/webkitpy/common/checkout/scm/scm.py:
(SCM.changed_files): Pass find_branch flag.
(SCM.create_patch): Ditto.
* Tools/Scripts/webkitpy/common/checkout/scm/scm_mock.py:
(MockSCM.changed_files): Pass find_branch flag.
(MockSCM.create_patch): Ditto.
* Tools/Scripts/webkitpy/common/checkout/scm/svn.py:
(SVN.changed_files): Pass find_branch flag.
(SVN.create_patch): Ditto.
* Tools/Scripts/webkitpy/style/main.py:
(CheckWebKitStyle.main): Check style from the branch point, not main.
* Tools/Scripts/webkitpy/tool/steps/abstractstep.py:
(AbstractStep): Create a patch from the branch point.
* Tools/Scripts/webkitpy/w3c/test_exporter_unittest.py:
(TestExporterTest.MockGit.create_patch): Pass find_branch flag.

Canonical link: https://commits.webkit.org/252647@main
